activerecord/test/cases/autosave_association_test.rb

```diff
@@ -2272,12 +2272,19 @@ def setup
     assert_equal(book_count_before_save, Book.count)
   end
 
-  def test_validations_still_fire_on_unchanged_association_with_custom_validation_context
+  test "validations still fire on unchanged association with custom validation context" do
     pirate = FamousPirate.create!(catchphrase: "Avast Ye!")
-    pirate.famous_ships.create!
+    FamousPirate.validates_associated :famous_ships
+    pirate.famous_ships.create!(name: "The Black Pearl")
 
     assert_predicate pirate, :valid?
-    assert_not pirate.valid?(:conference)
+    assert pirate.valid?(:conference)
+
+    pirate.reload
+    pirate.famous_ships.load
+
+    assert_predicate pirate, :valid?
+    assert pirate.valid?(:conference)
```

Comment:
I think what we want is for this to return false when the famous_ship don't have a name. For that we would need to add it to the database without validating